playbooks/operationalizing/vault_integration.adoc

```diff
@@ -0,0 +1,169 @@
+
+The following table summarises the threat model:
+
+|===
+|*Attack Vector*|*Vault Aware App*|*Legacy app*|*Openshift Secrets*
+
+|reading secrets at rest|ok|ok|no
+|reading secrets in transit|ok|ok|ok
+|reading secrets via the API|ok|ok|no
+|reading secrets in use from within the container (rsh)|ok|no|no
+|reading secrets in use from the node|ok|no|no
+|===
+
+== Installing the Vault infrastructure
+
+Clone this repository 
+----
+git clone https://github.com/raffaelespazzoli/credscontroller[https://github.com/raffaelespazzoli/credscontroller] 
+cd credscontroller
+----
+=== Create a new project
+----
+oc new-project vault-controller
+----
+=== Install Vault
+----
+oc adm policy add-scc-to-user anyuid -z default
+oc create configmap vault-config --from-file=vault-config=./openshift/vault-config.json
+oc create -f ./openshift/vault.yaml
+oc create route passthrough vault --port=8200 --service=vault
+----
+=== Initialize Vault
+----
+export VAULT_ADDR=https://`oc get route | grep -m1 vault | awk '{print $2}'`
+vault init -tls-skip-verify -key-shares=1 -key-threshold=1
+----
+=== Unseal Vault.
+
+You have to repeat this step every time you start vault. 
+Don't try to automate this step, this is manual by design. 
+You can make the initial seal stronger by increasing the number of keys. 
+We will assume that the KEYS environment variable contains the key necessary to unseal the vault and that ROOT_TOKEN contains the root token.
+
+For example: 
+----
+export KEYS=9++8KEDd72S3aGc0zaY9JW11tnQRDTEkCZWMHK2D0CM= 
+export ROOT_TOKEN=c30909da-a713-94bf-bf6e-46180ef79a64 
+vault unseal -tls-skip-verify $KEYS
+----
+
+=== Install Vault Controller
+----
+oc create secret generic vault-controller --from-literal vault-token=$ROOT_TOKEN
+oc adm policy add-cluster-role-to-user view system:serviceaccount:vault-controller:default
+oc create -f ./openshift/vault-controller.yaml
+----
+== Running the example
+
+At this point you are ready to start deploying pods that use this approach to inject secrets. 
+The repository comes with two example one for a https://github.com/raffaelespazzoli/credscontroller/tree/master/examples/spring-example[Vault aware app] and one for a https://github.com/raffaelespazzoli/credscontroller/tree/master/examples/spring-legacy-example[legacy app].  
+Here are the instructions to run the vault aware example. 
+In this first example a https://projects.spring.io/spring-boot/[spring boot] app uses the spring https://github.com/spring-cloud/spring-cloud-vault[cloud vault configuration plugin] to get part of its configuration from Vault. 
+The Init Container will write the unwrapped Vault token to a well know location, the app will use that token to authenticate with Vault and retrieve its credentials.
+
+=== Create the Vault policy
+
+We need to create a policy that will allow to access a specific subset of secrets. This is how you can have multiple application storing their secrets in Vault.
```

Comment:
before
`create a policy that will allow to access a specific`
after
`create a policy that will allow access to a specific`

`application` should also be plural `applications`